eclass/kernel-install.eclass

```diff
@@ -442,6 +595,41 @@ kernel-install_pkg_preinst() {
 	fi
 }
 
+# @FUNCTION: kernel-install_split_uki
+# @USAGE: <image path>
+# @DESCRIPTION:
+# Split a unified kernel image in the arch specific kernel image
+# and an initrd file. Does nothing unless KERNEL_IUSE_GENERIC_UKI is set.
+kernel-install_split_uki() {
+	[[ ${#} -eq 1 ]] || die "${FUNCNAME}: invalid arguments"
+	if [[ ${KERNEL_IUSE_GENERIC_UKI} ]]; then
+		local image_path=${1}
+		local uki="${image_path%/*}/uki.efi"
+		local initrd="${image_path%/*}/initrd"
+
+		if [[ -f ${uki} ]]; then
```

Comment:
I think this part is better in postinst too. Again, in -bin we know that UKI is there and we want an explicit error if it isn't for some reason.